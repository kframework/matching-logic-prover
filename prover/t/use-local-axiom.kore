// Tests that we can use local axioms.
symbol f(Int) : Int
claim \implies(\equals(f(1), 1), \equals(f(1), 1))
strategy intros H . apply-equation -> H at 0 by [] . reflexivity
