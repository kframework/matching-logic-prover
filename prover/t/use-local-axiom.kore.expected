<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom ax0 : \implies ( \equals ( f ( 1 , .Patterns ) , 1 ) , \equals ( f ( 1 , .Patterns ) , 1 ) )
    </declaration> <declaration>
      symbol f ( Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
